packages/vue/src/auth/entitlements.ts

```diff
@@ -9,36 +9,85 @@ import {
   AuthState,
 } from '@frontegg/redux-store';
 
+import { FeatureFlags, USE_ENTITLEMENTS_V2_ENDPOINT_FF } from '@frontegg/rest-api';
+
 import { authStateKey } from '../constants';
+import { useFronteggStore } from '../auth/mapAuthState';
 
-const getEntitlementsState = () => {
+/**
+ * @returns user state 
+ */
+const useGetUserState = () => {
   const authState = inject(authStateKey) as AuthState;
-  return authState.user?.entitlements as any;
+  return authState.user;
+};
+
+/**
+ * @returns true when need to use entitlements V2 API
+ */
+const useIsV2API = () => {
+  // TODO: check should it be immediate or we can recursively inject (check when app name change?)
+  const { appName } = useFronteggStore();
+  const [useEntitlementsV2] = FeatureFlags.getFeatureFlags([USE_ENTITLEMENTS_V2_ENDPOINT_FF], appName);
```

Comment:
I would make a generic useFeatureFlag hook that accepts the ff name from the outside and then it can be reusable

Reply:
will do